### PR TITLE
Bug fix 3.10/only remove subdir (#18079)

### PR DESCRIPTION
**Pull request by dothebart**
backport of https://github.com/arangodb/arangodb/pull/18079

### CODE REVIEW

**Review by alexbakharew — APPROVED** [review]
LGTM

File: tests/js/client/shell/shell-restore-integration.js [hunk 1]
```diff
@@ -1292,126 +1154,105 @@ function restoreIntegrationSuite() {
...
+        assertInstanceOf(arangodb.ArangoCollection, col);
+        assertEqual(colJson.parameters.name, col.name());
+        assertEqual(colJson.parameters.type, col.type());
+        if (isCluster) {
+          assertEqual(colJson.parameters.distributeShardsLike, col.properties().distributeShardsLike);
         }
-        fs.removeDirectoryRecursive(path, true);
-      } finally {
-        db._useDatabase("_system");
       }
```
Comment (mpoeter) [inline tests/js/client/shell/shell-restore-integration.js:1324]: There are a few places where you left the `_useDatabase` in a finally. Was this intentional? Should we keep the finally here, or can we remove the try-finally in the other cases as well?

**Review by mpoeter — APPROVED** [review]

File: tests/js/client/shell/shell-restore-integration.js (hunk 1 above)
Comment (dothebart) [reply tests/js/client/shell/shell-restore-integration.js:1324]: I guess not being in the system database is potentially bad for following up tests, hence if the test changes the database, it should change it back?
but probably `teardown` could be doing that as well, but I don't think we should do this in the backport.

Comment (mpoeter) [reply tests/js/client/shell/shell-restore-integration.js:1324]: Agreed about not doing this in the backport. I just wanted to point out that the changes are not fully consistent.

**Pull request merged**